File: llm_generated_code/Gemini/CPP/Level2/Q31.cpp
#include <iostream>

using namespace std;

int main() {
    int t;
    cin >> t;

    while (t--) {
        int a, b;
        cin >> a >> b;

        if (a == b) {
            cout << "DRAW" << endl;
        } else {
            cout << (a > b ? "ALICE" : "BOB") << endl;
        }
    }

    return 0;
}

File: llm_generated_code/Gemini/Python/Level2/Q31.py
def winner(a, b):
    if a == b:
        return "Alice" if a % 2 == 1 else "Bob"
    else:
        return "Alice" if a > b else "Bob"

def main():
    t = int(input())
    for _ in range(t):
        a, b = map(int, input().split())
        print(winner(a, b))

if __name__ == "__main__":
    main()
